future-apps/payroll/contracts/PayrollKit.sol

```diff
@@ -69,76 +70,169 @@ contract PayrollKit is KitBase {
-        address root = msg.sender;
-        address employer = msg.sender;
+    function newInstance()
+      public
+      returns (Kernel dao, Payroll payroll)
+    {
+      address root = msg.sender;
+      address employer = msg.sender;
 
-        dao = fac.newDAO(this);
-        ACL acl = ACL(dao.acl());
+      dao = fac.newDAO(this);
+      ACL acl = ACL(dao.acl());
 
-        PPFMock priceFeed = new PPFMock();
+      PPFMock priceFeed = new PPFMock();
 
-        MiniMeToken denominationToken = tokenFactory.createCloneToken(MiniMeToken(0), 0, "US Dollar", 18, "USD", true);
+      MiniMeToken denominationToken = newToken("USD Dolar", "USD");
 
-        acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
+      acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
 
-        Vault vault;
-        Finance finance;
-        (vault, finance, payroll) = deployApps(dao);
+      Vault vault;
+      Finance finance;
+      (vault, finance, payroll) = deployApps(dao);
 
-        finance.initialize(vault, financePeriodDuration);
-        payroll.initialize(finance, denominationToken, priceFeed, rateExpiryTime);
+      // Setup the permissions for the Finance App
+      setFinancePermissions(acl, finance, payroll, root);
 
-        // Payroll permissions
-        acl.createPermission(employer, payroll, payroll.ADD_EMPLOYEE_ROLE(), root);
-        acl.createPermission(employer, payroll, payroll.TERMINATE_EMPLOYEE_ROLE(), root);
-        acl.createPermission(employer, payroll, payroll.ALLOWED_TOKENS_MANAGER_ROLE(), root);
-        acl.createPermission(employer, payroll, payroll.SET_EMPLOYEE_SALARY_ROLE(), root);
-        acl.createPermission(employer, payroll, payroll.ADD_ACCRUED_VALUE_ROLE(), root);
-        acl.createPermission(root, payroll, payroll.CHANGE_PRICE_FEED_ROLE(), root);
-        acl.createPermission(root, payroll, payroll.MODIFY_RATE_EXPIRY_ROLE(), root);
+      // Setup the permissions for the vault
+      setVaultPermissions(acl, vault, finance, root);
 
-        // Finance permissions
-        acl.createPermission(payroll, finance, finance.CREATE_PAYMENTS_ROLE(), root);
+      // Payroll permissions
+      setPayrollPermissions(acl, payroll, root);
 
-        // Vault permissions
-        setVaultPermissions(acl, vault, finance, root);
+      vault.initialize();
+      finance.initialize(vault, financePeriodDuration);
+      payroll.initialize(finance, denominationToken, priceFeed, rateExpiryTime);
 
-        // EVMScriptRegistry permissions
-        // EVMScriptRegistry reg = EVMScriptRegistry(dao.getApp(dao.APP_ADDR_NAMESPACE(), EVMSCRIPT_REGISTRY_APP_ID));
-        // acl.createBurnedPermission(reg, reg.REGISTRY_ADD_EXECUTOR_ROLE());
-        // acl.createBurnedPermission(reg, reg.REGISTRY_MANAGER_ROLE());
+      // deployTokens(dao, finance, acl, root);
+      MiniMeToken token1 = deployAndDepositToken(dao, finance, acl, root, "Token 1", "TK1");
```

Comment:
What's the purpose of this? Is it just for testing? I think the kit should be intended to be deployed in mainnet. Maybe we could set whatever is needed for testing in an external script.